### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by KonstantinRyazantsev**
The job should get balances of all supported blockchains every day. The balances should be taken at specific time of the day. Result should be stored to a SQL DB.

### CODE REVIEW

File: src/Lykke.Job.BlockchainBalancesReport/Clients/BitsharesExploler/AccountHistoryResponse.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Newtonsoft.Json;
+
+namespace Lykke.Job.BlockchainBalancesReport.Clients.BitsharesExploler
```
Comment (cassade) [inline src/Lykke.Job.BlockchainBalancesReport/Clients/BitsharesExploler/AccountHistoryResponse.cs]: Typo - exploRer

File: src/Lykke.Job.BlockchainBalancesReport/Clients/Steemit/SteemetDeserializer.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace Lykke.Job.BlockchainBalancesReport.Clients.Steemit
+{
+    public static class SteemetDeserializer
```
Comment (cassade) [inline src/Lykke.Job.BlockchainBalancesReport/Clients/Steemit/SteemetDeserializer.cs]: Typo - Steemit

File: src/Lykke.Job.BlockchainBalancesReport/Clients/Steemit/SteemetDeserializer.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+                        ids.Add(id);
+                    }
+                }
+            }
+        }
+
+        private static decimal ParseSteemValue(string sourceValue)
+        {
+            return decimal.Parse(sourceValue.Replace(" STEEM", ""), CultureInfo.InvariantCulture);
```
Comment (cassade) [inline src/Lykke.Job.BlockchainBalancesReport/Clients/Steemit/SteemetDeserializer.cs]: Aside from STEEM there is SBD currency which could be transferred too. This line will throw on such transaction.

File: src/Lykke.Job.BlockchainBalancesReport/Reporting/AzureSqlReportRepository.cs [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+            );
+
+            var lastItem = items.Last();
+
+            foreach (var item in items)
+            {
+                sqlBuilder.AppendLine
+                (
+                    $@"
+                    date = {DateTimeValue(at)} and
```
Comment (cassade) [inline src/Lykke.Job.BlockchainBalancesReport/Reporting/AzureSqlReportRepository.cs:116]: Maybe it's better to use some tool like Dapper to make queries parameterized and avoid manual formatting?

**Review by cassade — DISMISSED** [review]

File: src/Lykke.Job.BlockchainBalancesReport/Clients/BitsharesExploler/AccountHistoryResponse.cs (hunk 1 above)
Comment (KonstantinRyazantsev) [reply src/Lykke.Job.BlockchainBalancesReport/Clients/BitsharesExploler/AccountHistoryResponse.cs]: fixed

File: src/Lykke.Job.BlockchainBalancesReport/Clients/Steemit/SteemetDeserializer.cs (hunk 2 above)
Comment (KonstantinRyazantsev) [reply src/Lykke.Job.BlockchainBalancesReport/Clients/Steemit/SteemetDeserializer.cs]: fixed

File: src/Lykke.Job.BlockchainBalancesReport/Clients/Steemit/SteemetDeserializer.cs (hunk 3 above)
Comment (KonstantinRyazantsev) [reply src/Lykke.Job.BlockchainBalancesReport/Clients/Steemit/SteemetDeserializer.cs]: It's ok for now. We'll fix it once first not STEEM will be found. Atm there are no SBD transactions on our hot wallet

File: src/Lykke.Job.BlockchainBalancesReport/Reporting/AzureSqlReportRepository.cs (hunk 4 above)
Comment (KonstantinRyazantsev) [reply src/Lykke.Job.BlockchainBalancesReport/Reporting/AzureSqlReportRepository.cs:116]: Afaik dapper doesn't support bulk operations. Dapper Plus supports them, but it's proprietary

**Review by cassade — APPROVED** [review]

**Pull request merged**